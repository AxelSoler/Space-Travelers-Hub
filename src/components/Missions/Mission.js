import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { joinMission, leaveMission } from '../../redux/missions/missions';

const Mission = (props) => {
  const {
    mission, description, id, join,
  } = props;
  const dispatch = useDispatch();
  const [isActive, setActive] = useState('false');

  const joinM = (e) => {
    const missionid = e.target.parentElement.id;
    dispatch(joinMission(missionid));
    setActive(!isActive);
  };

  const leaveM = (e) => {
    const missionid = e.target.parentElement.id;
    dispatch(leaveMission(missionid));
    setActive(!isActive);
  };

  return (
    <li id={id} className="tableLi">
      <h3 className="gridItem">{mission}</h3>
      <p className="gridItem descriptionP">{description}</p>
      <div className="gridItem">
        {join && (
          <h4>ACTIVE MEMBER</h4>
        )}
        {!join && (
          <h4>NOT A MEMBER</h4>
        )}
      </div>
      <div className="gridItem">
        {!join && (
          <button type="button" onClick={joinM}>Join Mission</button>
        )}
        {join && (
          <button type="button" onClick={leaveM}>Leave Mission</button>
        )}
      </div>
    </li>
  );
};

export default Mission;

Mission.propTypes = {
  mission: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  id: PropTypes.string.isRequired,
  join: PropTypes.bool,
};

Mission.defaultProps = { join: false };
