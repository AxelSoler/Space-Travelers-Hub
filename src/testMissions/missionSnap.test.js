import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import store from '../redux/configureStore';
import Mission from '../components/Missions/Mission';

test('a mission is rendering', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Mission
          key="1568412"
          mission="name of the mission"
          description="mission description"
          id="15"
          join
        />
      </Provider>,
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
