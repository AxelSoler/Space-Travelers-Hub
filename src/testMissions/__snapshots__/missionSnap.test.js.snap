// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`a mission is rendering 1`] = `
<li
  className="tableLi"
  id="15"
>
  <h3
    className="gridItem"
  >
    name of the mission
  </h3>
  <p
    className="gridItem descriptionP"
  >
    mission description
  </p>
  <h4
    className="gridItem"
  >
    ACTIVE MEMBER
  </h4>
  <button
    className="gridItem"
    onClick={[Function]}
    type="button"
  >
    Leave Mission
  </button>
</li>
`;
